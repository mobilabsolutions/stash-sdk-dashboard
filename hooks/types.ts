import { PspType } from 'components/types'

export interface CCConfig {
  ccExpiry: string
  ccHolderName: string
  ccMask: string
  ccType: string
}

export interface PayPalConfig {
  billingAgreementId: number
  deviceData: {
    correlation_id: number
  }
  nonce: string
}

export interface PersonalData {
  city: string
  country: string
  customerIP: string
  customerReference: string
  email: string
  firstName: string
  lastName: string
  street: string
  zip: number
}
export interface SepaConfig {
  bic: string
  iban: string
}
export enum PaymentMethod {
  SEPA = 'SEPA',
  PAY_PAL = 'PAY_PAL',
  CC = 'CC',
  GOOGLE_PAY = 'GOOGLE_PAY',
  APPLE_PAY = 'APPLE_PAY'
}
export interface Extra {
  ccConfig?: CCConfig
  payPalConfig?: PayPalConfig
  payload: string
  paymentMethod: PaymentMethod
  personalData: PersonalData
  sepaConfig?: SepaConfig
}
export enum TransactionAction {
  AUTH = 'AUTH',
  PREAUTH = 'PREAUTH',
  REVERSAL = 'REVERSAL',
  REFUND = 'REFUND',
  ADDITIONAL = 'ADDITIONAL',
  CHARGEBACK = 'CHARGEBACK',
  'CHARGEBACK-REVERSAL' = 'CHARGEBACK-REVERSAL',
  CAPTURE = 'CAPTURE'
}

export enum TransactionStatus {
  SUCCESS = 'SUCCESS',
  PENDING = 'PENDING',
  FAIL = 'FAIL'
}

export interface PspConfig {
  accountId: string
  country: string
  currency: string
  default: true
  key: string
  locale: string
  merchantId: string
  portalId: string
  privateKey: string
  publicKey: string
  sandboxMerchantId: string
  sandboxPrivateKey: string
  sandboxPublicKey: string
  type: PspType
  urlPrefix: string
}

export interface TimeAction {
  action: TransactionAction
  amount: number
  createdDate: string
  reason: string
  status: TransactionStatus
}

export interface TransactionDetails {
  action: TransactionAction
  aliasId: string
  amount: number
  createdDate: string
  currencyId: string
  merchantCustomerId: string
  merchantId: string
  merchantTransactionId: string
  paymentInfo: {
    extra: Extra
    pspConfig: PspConfig
  }
  paymentMethod: PaymentMethod
  pspTestMode: true
  reason: string
  status: TransactionStatus
  timelineInfo: Array<TimeAction>
  transactionId: string
}
