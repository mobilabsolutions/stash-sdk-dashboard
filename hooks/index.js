export { useLocalization } from './use_localization'
export { useSettings } from './use_settings'
export { useTransactions } from './use_transactions'
