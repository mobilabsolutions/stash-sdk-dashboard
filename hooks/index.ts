export { useClientRect } from './use_client_rects'
export { useLocalization } from './use_localization'
export { useKeyDown } from './use_key_down'
export { useSettings } from './use_settings'
export { useApi } from './use_api'
export { useTransactions } from './use_transactions'
