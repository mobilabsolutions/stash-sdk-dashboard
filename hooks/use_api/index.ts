import { useCallback } from 'react'
import { useNextContext } from '../use_next_context'

const BACKEND_HOST =
  typeof window !== 'undefined'
    ? ''
    : process.env.API_UPSTREAM || 'https://pd.mblb.net'

const apiCall = (
  token: string,
  refresh: () => Promise<void>,
  method: string,
  path: string,
  content: object = null
) => {
  const request: any = token
    ? {
        method,
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${token}`
        },
        credentials: 'include'
      }
    : {
        method,
        headers: {
          'Content-Type': 'application/json'
        }
      }

  if (content) {
    request.body = JSON.stringify(content)
  }

  const url = `${BACKEND_HOST}${path}`
  let didRefresh = false

  const processApiCall = (request: any) => {
    return new Promise((resolve, reject) => {
      return fetch(url, request)
        .then(response => {
          if (response.status >= 500) {
            const error: any = new Error('Api Server Error')
            error.statusCode = response.status
            error.statusText = response.statusText
            return reject(error)
          }

          if (response.status === 401 && !didRefresh) {
            return refresh().then(accessToken => {
              didRefresh = true
              return processApiCall({
                method,
                headers: {
                  'Content-Type': 'application/json',
                  Authorization: `Bearer ${accessToken}`
                },
                credentials: 'include'
              })
            })
          }
          if (response.status >= 400) {
            return response
              .json()
              .then(json => {
                const error: any = new Error('Api User Error')
                error.statusCode = response.status
                error.statusText = response.statusText
                error.reason = json.reason
                error.data = json.data
                reject(error)
              })
              .catch(error => {
                error.statusCode = response.status
                error.statusText = response.statusText
                reject(error)
              })
          }

          if (response.status == 204) {
            return resolve({ result: null, statusCode: response.status })
          }

          response
            .json()
            .then(result => resolve({ result, statusCode: response.status }))
            .catch(reject)
        })
        .catch(reject)
    })
  }

  return processApiCall(request)
}

export const useApi = () => {
  const { cookies, setCookie } = useNextContext()

  const token = cookies['__token']
  const refreshToken = cookies['__refreshToken']

  const refresh = useCallback(
    () =>
      apiCall(refreshToken, null, 'POST', '/api/v1/token/refresh', null).then(
        ({ result: { refreshToken, accessToken } }) => {
          setCookie('__token', accessToken)
          setCookie('__refreshToken', refreshToken)

          return accessToken
        }
      ),
    [refreshToken, setCookie]
  )

  const get = useCallback(
    (path: string) => apiCall(token, refresh, 'GET', path),
    [refresh, token]
  )
  const put = useCallback(
    (path: string, content: object) =>
      apiCall(token, refresh, 'PUT', path, content),
    [refresh, token]
  )
  const patch = useCallback(
    (path: string, content: object) =>
      apiCall(token, refresh, 'PATCH', path, content),
    [refresh, token]
  )
  const post = useCallback(
    (path: string, content: object) =>
      apiCall(token, refresh, 'POST', path, content),
    [refresh, token]
  )
  const del = useCallback(
    (path: string) => apiCall(token, refresh, 'DELETE', path),
    [refresh, token]
  )

  const login = useCallback(
    (username: string, password: string) =>
      apiCall(null, null, 'POST', '/api/v1/token', { username, password }).then(
        ({ result: { refreshToken, accessToken } }) => {
          setCookie('__token', accessToken)
          setCookie('__refreshToken', refreshToken)
        }
      ),
    [setCookie]
  )

  const logout = useCallback(() => {
    setCookie('__token', null)
    setCookie('__refreshToken', null)
  }, [setCookie])

  return { get, put, patch, post, del, token, login, logout, refresh }
}
