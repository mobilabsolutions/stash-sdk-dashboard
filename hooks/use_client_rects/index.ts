import { useEffect, useState, MutableRefObject } from 'react'

export const useClientRect = (ref: MutableRefObject<any>) => {
  const getClientRect = () => {
    if (!ref || !ref.current) return {}

    const clientRects = ref.current.getClientRects()
    return clientRects.length > 0 ? clientRects[0] : {}
  }

  const [state, setState] = useState(getClientRect())
  const updateState = () => setState(getClientRect())

  useEffect(() => {
    updateState()

    window.addEventListener('resize', updateState)
    return () => window.removeEventListener('resize', updateState)
  }, [ref.current])

  return state
}
