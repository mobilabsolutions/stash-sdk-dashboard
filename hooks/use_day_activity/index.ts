import { useState, useEffect } from 'react'
import { TodaysActivity as Activity } from '../../types'
import { useApi } from '../'
import { Moment, defaultFormatUtc } from 'moment'

interface ActivityState {
  activities: Activity[]
  loading: boolean
  error: any
}

interface ApiResponse {
  transactions: {
    amount: number
    hour: number
  }[]
}

function initialValues(): ActivityState {
  return {
    activities: [],
    loading: false,
    error: null
  }
}

export default (date: Moment) => {
  const [state, setState] = useState(initialValues())
  const { get, merchantId } = useApi()
  const load = () => {
    setState(prevState => ({ ...prevState, loading: true }))
    get(
      `/api/v1/home/${encodeURIComponent(merchantId)}/activity?date=${date
        .clone()
        .hours(0)
        .minutes(0)
        .format(defaultFormatUtc)}`
    )
      .then((response: { result: ApiResponse }) => {
        const { transactions = [] } = response.result

        return setState({
          activities: transactions.map(trans => ({
            time: trans.hour.toString(),
            amount: trans.amount
          })),
          loading: false,
          error: null
        })
      })
      .catch((error: any) => {
        setState(prev => ({ ...prev, loading: false, error }))
      })
  }

  useEffect(load, [date])

  return {
    ...state,
    load
  }
}
