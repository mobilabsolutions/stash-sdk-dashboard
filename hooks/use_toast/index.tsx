import React, { useContext, useState, useCallback } from 'react'

import styled, { keyframes } from '../../components/styled'

const fadeInUp = keyframes`
  from {
    opacity: 0;
    transform: translate3d(0, 100%, 0);
  }
  to {
    opacity: 1;
    transform: none;
  }
`

const ToastContainer = styled.div`
  position: absolute;
  overflow: hidden;
  z-index: 999999999999;
  max-height: calc(100vh - 10px);
  text-align: right;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  bottom: 100px;
`

const Toast = styled.div`
  font-family: ${props => props.theme.font};
  display: flex;
  align-items: center;
  text-align: center;
  padding: 5px 15px;
  white-space: pre-line;
  min-height: 50px;
  margin-bottom: 15px;
  border-radius: 5px;
  animation-name: ${fadeInUp};
  animation-duration: 1s;
  animation-fill-mode: both;
  transform: translateX(-50%);
`

const toastContext = React.createContext(null)
toastContext.displayName = 'NextContext'

export const ToastProvider = ({ children }) => {
  const [toasts, setToasts] = useState([])

  const success = useCallback((message: string, timer: number) => {
    console.log(message)
    setToasts(prevToasts => [
      ...prevToasts,
      { message, type: 'success', valid: Date.now() + timer }
    ])
  }, [])

  const providerValue = {
    toasts,
    success
  }

  return (
    <toastContext.Provider value={providerValue}>
      {children}
      <ToastContainer>
        {toasts.map((toast, index) => (
          <Toast key={index}>{toast.message}</Toast>
        ))}
      </ToastContainer>
    </toastContext.Provider>
  )
}

export const useToast = () => {
  const { success } = useContext(toastContext)

  return { success }
}
