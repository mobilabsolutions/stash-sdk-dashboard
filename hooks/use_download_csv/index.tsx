import FileSaver from 'file-saver'
import { useApi } from '../use_api'
import { useState } from 'react'

const getInitial = () => ({
  downloading: false,
  error: false
})

const type = 'text/csv;charset=ISO-8859-1'

export default (url?: string, filename?: string) => {
  const [state, setstate] = useState(getInitial())
  const { getRaw } = useApi()

  async function download(customUrl?: string, customFilename?: string) {
    setstate({ downloading: true, error: false })
    try {
      const { result } = await getRaw(!!customUrl ? customUrl : url)
      const text = await result.text()
      var blob = new Blob([text], { type })
      FileSaver.saveAs(blob, !!customFilename ? customFilename : filename)
      setstate({ downloading: false, error: false })
    } catch (e) {
      setstate({ downloading: false, error: true })
    }
  }

  return {
    ...state,
    download
  }
}
