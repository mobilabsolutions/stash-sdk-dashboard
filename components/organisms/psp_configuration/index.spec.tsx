import { testRender, deepRender } from '../../../test_utils'
import PspCreateForm from './psp_create_form'
import { fireEvent, queryByAttribute } from 'react-testing-library'
import { ToastProvider } from '../../../hooks/use_toast'
import { cleanup } from 'react-testing-library'

const ToastWrapper = Cmp => p => (
  <ToastProvider>
    <Cmp {...p} />
  </ToastProvider>
)

describe('PspCreateForm', () => {
  afterEach(cleanup)
  test('should render just the button first', () => {
    const FormWithToast = ToastWrapper(PspCreateForm)
    testRender(FormWithToast, {
      onCreatePsp: jest.fn(),
      pspsToCreate: [
        {
          label: 'BS PayOne',
          value: 'BS_PAYONE'
        },
        {
          label: 'Adyen',
          value: 'ADYEN'
        },
        {
          label: 'Braintree',
          value: 'BRAINTREE'
        }
      ]
    })
  })

  test('should render', () => {
    const onCreatePsp = jest.fn()
    const FormWithToast = ToastWrapper(PspCreateForm)
    const { getByText, baseElement } = deepRender(FormWithToast, {
      onCreatePsp,
      pspsToCreate: [
        {
          label: 'BS PayOne',
          value: 'BS_PAYONE'
        },
        {
          label: 'Adyen',
          value: 'ADYEN'
        },
        {
          label: 'Braintree',
          value: 'BRAINTREE'
        }
      ]
    })
    fireEvent.click(getByText('Add a new PSP configuration'))
    expect(baseElement.firstChild).toMatchSnapshot(
      'the "select" and "cancel" button after click "+"'
    )

    //Open select menu typing on input
    fireEvent.change(
      queryByAttribute('id', baseElement, 'react-select-2-input'),
      {
        target: { value: 'a' }
      }
    )

    //Click first option
    fireEvent.click(
      queryByAttribute('id', baseElement, 'react-select-2-option-0')
    )

    expect(baseElement.firstChild).toMatchSnapshot(
      'the form for the creation of the selected PSP'
    )
  })
})
