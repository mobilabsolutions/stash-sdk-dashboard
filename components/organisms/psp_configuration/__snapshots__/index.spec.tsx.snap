// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PspCreateForm should render just the button first 1`] = `
Array [
  <div
    className="psp_create_form__Container-sc-11to5vc-0 kWdnas"
  >
    <button
      className="primary__Button-sc-1lwi4g7-0 kLABFK"
      disabled={false}
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      <span
        className="add-plus"
      >
        <svg
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 7h6a1 1 0 0 1 0 2H9v6a1 1 0 0 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 2 0v6z"
            fill="#FFF"
            fillRule="evenodd"
          />
        </svg>
        Add a new PSP configuration
      </span>
    </button>
  </div>,
  <div
    className="use_toast__ToastContainer-sc-1wws89f-0 glXDBM"
  />,
]
`;

exports[`PspCreateForm should render: the "select" and "cancel" button after click "+" 1`] = `
<div>
  <div
    class="psp_create_form__Container-sc-11to5vc-0 kWdnas"
  >
    <div
      class="select-cancel-container"
    >
      <div
        class="psp-select css-chbh3k-container"
      >
        <div
          class=" css-vr5vfp-control"
        >
          <div
            class=" css-1hwfws3"
          >
            <div
              class=" css-1y2mitp-placeholder"
            >
              Select a PSP
            </div>
            <div
              class="css-pe2t16"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1wy0on6"
          >
            <span />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <button
        class="secondary__Button-j7mjyk-0 kgyVVR"
        type="button"
      >
        <span>
          Cancel
        </span>
      </button>
    </div>
  </div>
  <div
    class="use_toast__ToastContainer-sc-1wws89f-0 glXDBM"
  />
</div>
`;

exports[`PspCreateForm should render: the form for the creation of the selected PSP 1`] = `
<div>
  <div
    class="psp_create_form__Container-sc-11to5vc-0 kWdnas"
  >
    <div
      class="select-cancel-container"
    >
      <div
        class="psp-select css-chbh3k-container"
      >
        <div
          class=" css-vr5vfp-control"
        >
          <div
            class=" css-1hwfws3"
          >
            <div
              class=" css-1nzxrjy-singleValue"
            >
              BS PayOne
            </div>
            <div
              class="css-pe2t16"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 0; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1wy0on6"
          >
            <span />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="psp-create-form"
    >
      <div>
        <h3
          class="h3-sc-1przhcr-0 bqBPGG"
        >
          BS Payone
        </h3>
        <form
          class="psp_form__Form-u54q1b-0 QkbWZ"
        >
          <h3
            class="h3-sc-1przhcr-0 bspayone__H4-sc-9o178t-0 jCMFDG"
          >
            Merchant ID
          </h3>
          <div
            class="field_wrapper-n32k2e-0 irXinp"
          >
            <div
              class="input_wrapper-sc-15hmhfy-0 duFQAu"
            >
              <div
                class="icon_wrapper-sc-1jgazwq-0 VoEeG"
              >
                <svg
                  class="enable"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2a7.2 7.2 0 0 1-6-3.22c.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08a7.2 7.2 0 0 1-6 3.22z"
                    fill="#D6D9DB"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <input
                class="input__HtmlInput-rcagn3-0 iMczqN"
                name="merchantId"
                placeholder="Merchant ID"
                type="text"
                value=""
              />
            </div>
          </div>
          <h3
            class="h3-sc-1przhcr-0 bspayone__H4-sc-9o178t-0 jCMFDG"
          >
            Portal ID
          </h3>
          <div
            class="field_wrapper-n32k2e-0 irXinp"
          >
            <div
              class="input_wrapper-sc-15hmhfy-0 duFQAu"
            >
              <div
                class="icon_wrapper-sc-1jgazwq-0 VoEeG"
              >
                <svg
                  class="enable"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2a7.2 7.2 0 0 1-6-3.22c.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08a7.2 7.2 0 0 1-6 3.22z"
                    fill="#D6D9DB"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <input
                class="input__HtmlInput-rcagn3-0 iMczqN"
                name="portalId"
                placeholder="Portal ID"
                type="text"
                value=""
              />
            </div>
          </div>
          <h3
            class="h3-sc-1przhcr-0 bspayone__H4-sc-9o178t-0 jCMFDG"
          >
            Key
          </h3>
          <div
            class="field_wrapper-n32k2e-0 irXinp"
          >
            <div
              class="input_wrapper-sc-15hmhfy-0 duFQAu"
            >
              <div
                class="icon_wrapper-sc-1jgazwq-0 VoEeG"
              >
                <svg
                  class="enable"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M13.65 11A5.99 5.99 0 0 0 8 7c-3.31 0-6 2.69-6 6s2.69 6 6 6a5.99 5.99 0 0 0 5.65-4H18v4h4v-4h2v-4H13.65zM8 15c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                    fill="#D6D9DB"
                    fill-rule="evenodd"
                    transform="rotate(-45 7.879 15.121)"
                  />
                </svg>
              </div>
              <input
                class="input__HtmlInput-rcagn3-0 iMczqN"
                name="key"
                placeholder="Key"
                type="text"
                value=""
              />
            </div>
          </div>
          <h3
            class="h3-sc-1przhcr-0 bspayone__H4-sc-9o178t-0 jCMFDG"
          >
            Account ID
          </h3>
          <div
            class="field_wrapper-n32k2e-0 irXinp"
          >
            <div
              class="input_wrapper-sc-15hmhfy-0 duFQAu"
            >
              <div
                class="icon_wrapper-sc-1jgazwq-0 VoEeG"
              >
                <svg
                  class="enable"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2a7.2 7.2 0 0 1-6-3.22c.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08a7.2 7.2 0 0 1-6 3.22z"
                    fill="#D6D9DB"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <input
                class="input__HtmlInput-rcagn3-0 iMczqN"
                name="accountId"
                placeholder="Account ID"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="bspayone__ButtonContainer-sc-9o178t-1 kpJOqC"
          >
            <button
              class="secondary__Button-j7mjyk-0 kgyVVR"
              type="button"
            >
              <span>
                Cancel
              </span>
            </button>
            <button
              class="primary__Button-sc-1lwi4g7-0 kLABFK"
              disabled=""
              type="submit"
            >
              <span>
                Save
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    class="use_toast__ToastContainer-sc-1wws89f-0 glXDBM"
  />
</div>
`;
